react/ThumbnailItem.js

```diff
@@ -0,0 +1,39 @@
+import React, { Component } from 'react'
+import PropTypes from 'prop-types'
+
+class ThumbnailItem extends Component {
+  handleClick(event) {
```

Comment:
Usar arrow function.

```js
handleClick = event => {
  
}
```

Desta forma vc não precisa usar `this.handleClick.bind(this)` no `render`, pode apenas usar `this.handleClick`.